programs/locker/src/instructions/v3/claim.rs

```diff
@@ -0,0 +1,224 @@
+        seeds = [
+            b"claim_status".as_ref(),
+            recipient.key().to_bytes().as_ref(),
+            escrow.key().to_bytes().as_ref()
+        ],
+        bump,
+        space = 8 + ClaimStatus::INIT_SPACE,
+        payer = recipient
+    )]
+    pub claim_status: Account<'info, ClaimStatus>,
+
+    /// Escrow.
+    #[account(
+        mut,
+        has_one = token_mint,
+        constraint = escrow.load()?.cancelled_at == 0 @ LockerError::AlreadyCancelled
+    )]
+    pub escrow: AccountLoader<'info, VestingEscrowV3>,
+
+    /// Mint.
+    pub token_mint: Box<InterfaceAccount<'info, Mint>>,
+
+    /// Escrow Token Account.
+    #[account(
+        mut,
+        associated_token::mint = token_mint,
+        associated_token::authority = escrow,
+        associated_token::token_program = token_program
+    )]
+    pub escrow_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// Recipient.
+    #[account(mut)]
+    pub recipient: Signer<'info>,
+
+    /// Recipient Token Account.
+    #[account(
+        mut,
+        constraint = recipient_token.key() != escrow_token.key() @ LockerError::InvalidRecipientTokenAccount
+    )]
+    pub recipient_token: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// Memo program.
+    pub memo_program: Program<'info, Memo>,
+
+    /// Token program.
+    pub token_program: Interface<'info, TokenInterface>,
+    /// system program.
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handle_claim_v3<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, ClaimV3<'info>>,
+    params: &ClaimV3Params,
+    remaining_accounts_info: Option<RemainingAccountsInfo>,
+) -> Result<()> {
+    let claim_status = &mut ctx.accounts.claim_status;
+    let mut escrow = ctx.accounts.escrow.load_mut()?;
+
+    params.verify_recipient(ctx.accounts.recipient.key(), escrow.root)?;
+
+    let amount = params.get_claim_amount(claim_status.total_claimed_amount)?;
+    escrow.total_claimed_amount += amount;
+    drop(escrow);
+
+    // Process remaining accounts
+    let mut remaining_accounts = &ctx.remaining_accounts[..];
+    let parsed_transfer_hook_accounts = match remaining_accounts_info {
+        Some(info) => parse_remaining_accounts(
+            &mut remaining_accounts,
+            &info.slices,
+            &[AccountsType::TransferHookEscrow],
+        )?,
+        None => ParsedRemainingAccounts::default(),
+    };
+
+    transfer_to_user_v3(
+        &ctx.accounts.escrow,
+        &ctx.accounts.token_mint,
+        &ctx.accounts.escrow_token.to_account_info(),
+        &ctx.accounts.recipient_token,
+        &ctx.accounts.token_program,
+        Some(MemoTransferContext {
+            memo_program: &ctx.accounts.memo_program,
+            memo: TRANSFER_MEMO_CLAIM_VESTING.as_bytes(),
+        }),
+        amount,
+        parsed_transfer_hook_accounts.transfer_hook_escrow,
+    )?;
+
+    // update claim status
+    claim_status.total_claimed_amount += amount;
```

Comment:
Pls use safe math, and could you push those operations inside ClaimStatus state?

Reply:
i fixed it.